x-pack/legacy/plugins/siem/index.ts

```diff
@@ -125,17 +123,36 @@ export const siem = (kibana: any) => {
       mappings: savedObjectMappings,
     },
     init(server: Server) {
-      const newPlatform = ((server as unknown) as KbnServer).newPlatform;
-      if (server.plugins.alerting != null) {
-        const type = signalsAlertType({
-          logger: newPlatform.coreContext.logger.get('plugins', APP_ID),
-        });
-        if (isAlertExecutor(type)) {
-          server.plugins.alerting.setup.registerType(type);
-        }
-      }
-      server.injectUiAppVars('siem', async () => server.getInjectedUiAppVars('kibana'));
-      initServerWithKibana(server);
+      const {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
+        newPlatform,
+        plugins,
+        route,
+        savedObjects,
+      } = server;
+
+      const { logger } = newPlatform.coreContext;
+
+      const serverFacade = {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
+        newPlatform: {
+          coreContext: { logger },
+          env: newPlatform.env,
+        },
+        plugins: { alerting: plugins.alerting, xpack_main: plugins.xpack_main },
+        route: route.bind(server),
+        savedObjects,
+      };
+
+      plugin({ logger }).setup({}, {});
+
+      initServerWithKibana(serverFacade);
```

Comment:
you can move this function into your `Plugin` class. Then you could instantiate your plugin with `plugin(newPlatform.coreContext).setup(newPlatform.setup.core, newPlatform.setup.plugins, ServerFacade).start(newPlatform.start.core, newPlatform.start.plugins);` Here I'm passing in `ServerFacade` as a third parameter that we use just for legacy dependencies, once moved to the NP this third parameter would go completely away.

You would then want to move `initServerWithKibana` into your plugin so that everything (even your legacy stuff) is driven from within your plugin.

Reply:
Done. I haven't implemented `start` and I'm simply invoking `initServerWithKibana` from the plugin for now, but it will eventually be inlined.